extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -268,7 +268,8 @@ public void onFailure(Throwable t)
         }
       }
       catch (InterruptedException | ExecutionException | TimeoutException e) {
-        if (!future.isDone() && !future.cancel(true) && !future.isDone()) {
+        executorService.shutdown();
+        if (!future.isDone() && !future.cancel(false)) {
```

Comment:
@gvsmirnov in the similar situation I just removed this check: https://github.com/druid-io/druid/pull/3609/commits/7effa7acc2fc52644f1957fea187693bb0ccb591 since it's anyway not possible (according to the `Future` specification) to "fail to cancel" for any reason other than the future already failed or cancelled, or finished itself normally. So this is a check of JDK implementation sanity, that is excessive IMHO.